src/OidcClient.php

```diff
@@ -238,6 +238,33 @@ public function retrieveUserInfo(OidcTokens $tokens): OidcUserData
     return new OidcUserData($data);
   }
 
+  public function getIntrospection(OidcTokens $tokens): array
+  {
+    $headers = [];
+    if (in_array('client_secret_basic', $this->getTokenEndpointAuthMethods())) {
+      $headers = [
+        'Authorization: Basic ' . base64_encode(urlencode($this->clientId) . ':' . urlencode($this->clientSecret)),
+      ];
+    }
+
+    $jsonData = $this->urlFetcher->fetchUrl(
+      $this->getConfigurationValue('introspection_endpoint'),
+      ['token' => $tokens->getAccessToken()],
+      $headers,
+    );
+    $jsonData = mb_convert_encoding($jsonData, 'UTF-8');
+
+    // Read the data
+    $data = json_decode($jsonData, true);
+
+    // Check data due
+    if (!is_array($data)) {
+      throw new OidcException('Error from the introspection endpoint.');
+    }
+
+    return $data;
```

Comment:
I'd like to see a model for this, as the response from the introspection endpoint is strictly defined: https://datatracker.ietf.org/doc/html/rfc7662#section-2.2. 

Best is something similar to the OidcUserData model.